client/services/FeedbackGeneratorService.js

```diff
@@ -373,21 +447,58 @@ tie.factory('FeedbackGeneratorService', [
        * @returns {Feedback}
        */
       getFeedback: function(tasks, codeEvalResult, rawCodeLineIndexes) {
+        // Reset all counters but consecutiveSameErrorCounter.
+        _resetCounters('consecutiveSameError');
+
+        // If the user recieves the same error message increment the same error
+        // counter.
+        if (previousErrorString === codeEvalResult.getErrorString()) {
+          consecutiveSameErrorCounter++;
+        }
+
         var feedback = _getFeedbackWithoutReinforcement(
           tasks, codeEvalResult, rawCodeLineIndexes);
         if (tasks.length > 0 && !codeEvalResult.getErrorString()) {
           feedback.setReinforcement(
             ReinforcementGeneratorService.getReinforcement(
               tasks[tasks.length - 1], codeEvalResult));
         }
+
+        // If the same error counter reaches the threshold, prompt the user to
+        // look at the primer.
+        if (consecutiveSameErrorCounter === UNFAMILIARITY_THRESHOLD) {
+          feedback.appendTextParagraph(_getUnfamiliarLanguageFeedback());
+          consecutiveSameErrorCounter = 0;
```

Comment:
Hm, why are we doing this here? Could the user be getting the same error because their code works fine, but keeps printing out the wrong solution? In such a case, is looking at the primer actually helpful?